MobileFuse Adapter: Remove tagid_src and pub_id params

tomaszbmf opened the pull request:
This PR removes the `tagid_src` parameter, which was previously used to signal that the publisher is providing their own ID instead of using ours, and the `pub_id` query parameter, which was only necessary when `tagid_src` was present. After confirming that publishers do not utilize these parameters, we determined both are unnecessary. Removing them will simplify the setup and reduce confusion for our users.

### 🔧 Type of changes
- [x] bid adapter update
- [x] configuration

### ✨ What's the context?
`tagid_src` is effectively not used, `pub_id` is only necessary when `tagid_src` is present

### 🧠 Rationale behind the change
Removing it will help simplify the setup and reduce confusion for our users.

### 🧪 Test plan
we don't see requests with `tagid_src` query param

### 🏎 Quality check
- [x] Are your changes following [our code style guidelines](https://github.com/prebid/prebid-server-java/blob/master/docs/developers/code-style.md)?

AntoxaAntoxic commented on src/main/java/org/prebid/server/bidder/mobilefuse/MobilefuseBidder.java:
```diff
@@ -99,8 +99,7 @@ private ObjectNode parseSkadn(ObjectNode impExt) {
     private String makeUrl(ExtImpMobilefuse extImp) {
-        final String baseUrl = endpointUrl + Objects.toString(extImp.getPublisherId(), "0");
-        return "ext".equals(extImp.getTagidSrc()) ? baseUrl + "&tagid_src=ext" : baseUrl;
+        return endpointUrl;
```
looks like you don't the following part of code as well
```java
        final String endpoint = request.getImp().stream()
                .map(this::parseImpExt)
                .filter(Objects::nonNull)
                .findFirst()
                .map(this::makeUrl)
                .orElse(null);
```

please clean up redundant code

tomaszbmf replied:
Thanks for pointing that out! I've removed the redundant code as suggested.

To ensure stability after removing the `.map(this::parseImpExt).filter(Objects::nonNull) `chain, I added a null check in the `modifyImp(Imp imp)` method. This prevents a potential NullPointerException in cases where `parseImpExt` could return null.

Let me know if there's anything else you'd like me to update!

CTMBNara commented on src/main/java/org/prebid/server/bidder/mobilefuse/MobilefuseBidder.java:
```diff
@@ -75,6 +64,10 @@ private Imp modifyImp(Imp imp) {
         }
 
         final ExtImpMobilefuse impExt = parseImpExt(imp);
+        if (impExt == null) {
+            return null;
+        }
```
No need for this. If there is no `impExt` then imp will be filtered out for that bidder

CTMBNara replied:
Also, I see that in GO they fail on invalid `impExt` instead of ignoring errors

tomaszbmf replied:
Thanks! I've removed the check as suggested.

CTMBNara commented:
@tomaszbmf Please, change the logic so that the adapter fails on invalid `imp.ext`, like in GO, instead of ignoring errors

https://github.com/prebid/prebid-server-java/pull/3915#discussion_r2091430109

tomaszbmf commented:
> @tomaszbmf Please, change the logic so that the adapter fails on invalid `imp.ext`, like in GO, instead of ignoring errors
> 
> [#3915 (comment)](https://github.com/prebid/prebid-server-java/pull/3915#discussion_r2091430109)

I've updated the code to reintroduce the `"Invalid ExtImpMobilefuse value"` error when no valid `ext` is present in any of the request `imp` objects.

Previously, the presence of `ext` was checked indirectly as part of building the `endpoint` from `ext` values. Since the endpoint is no longer constructed based on those values, I had initially removed that logic. However, based on feedback, I’ve added back the validation in a simplified form:

```
final boolean hasExt = request.getImp().stream()
    .map(this::parseImpExt)
    .anyMatch(Objects::nonNull);

if (!hasExt) {
    return Result.withError(BidderError.badInput("Invalid ExtImpMobilefuse value"));
}
```

Reintroduces the "Invalid ExtImpMobilefuse value" error when no `imp` in the request contains a valid `ext` object. This preserves the original validation logic that was previously removed when the endpoint construction stopped relying on `ext`.

This change retains the updated endpoint handling logic and does not require any unit test modifications compared to the `main` branch.

CTMBNara reviewed (changes requested):
First, the updated code still doesn't comply with GO logic: you need to collect all errors until valid `ext`
```
	mobileFuseExtension, errs := getFirstMobileFuseExtension(bidRequest)
	if errs != nil {
		return nil, errs
	}
```
```
		err := jsonutil.Unmarshal(imp.Ext, &bidder_imp_extension)

		if err != nil {
			errs = append(errs, err)
			continue
		}

		err = jsonutil.Unmarshal(bidder_imp_extension.Bidder, &mobileFuseImpExtension)

		if err != nil {
			errs = append(errs, err)
			continue
		}
```

Also, to simplify the adapter code and bring it closer to the GO logic, I suggest changing it a little as follows:
```
    @Override
    public Result<List<HttpRequest<BidRequest>>> makeHttpRequests(BidRequest request) {
        final List<Imp> modifiedImps = new ArrayList<>();
        final List<BidderError> errors = new ArrayList<>();

        for (Imp imp : request.getImp()) {
            try {
                if (!isValidImp(imp)) {
                    continue;
                }

                final ExtImpMobilefuse extImp = parseImpExt(imp);
                modifiedImps.add(modifyImp(imp, extImp));
            } catch (PreBidException e) {
                errors.add(BidderError.badInput(e.getMessage()));
            }
        }

        if (!errors.isEmpty()) {
            return Result.withErrors(errors);
        }

        final BidRequest modifiedRequest = request.toBuilder().imp(modifiedImps).build();
        return Result.withValue(BidderUtil.defaultRequest(modifiedRequest, endpointUrl, mapper));
    }

    private static boolean isValidImp(Imp imp) {
        return imp.getBanner() != null || imp.getVideo() != null || imp.getXNative() != null;
    }

    private ExtImpMobilefuse parseImpExt(Imp imp) {
        try {
            return mapper.mapper().convertValue(imp.getExt(), MOBILEFUSE_EXT_TYPE_REFERENCE).getBidder();
        } catch (IllegalArgumentException e) {
            throw new PreBidException(e.getMessage());
        }
    }

    private Imp modifyImp(Imp imp, ExtImpMobilefuse extImp) {
        final ObjectNode skadn = parseSkadn(imp.getExt());
        return imp.toBuilder()
                .tagid(Objects.toString(extImp.getPlacementId(), "0"))
                .ext(skadn != null ? mapper.mapper().createObjectNode().set(SKADN_PROPERTY_NAME, skadn) : null)
                .build();
    }

    private ObjectNode parseSkadn(ObjectNode impExt) {
        try {
            return mapper.mapper().convertValue(impExt.get(SKADN_PROPERTY_NAME), ObjectNode.class);
        } catch (IllegalArgumentException e) {
            throw new PreBidException(e.getMessage());
        }
    }
```

tomaszbmf commented:
Thanks for the clarification and the code suggestion! 🙏 

I’ve now pushed the updated changes incorporating your snippet (with a minor adjustment). Initially, I was aiming to keep the scope minimal and focused strictly on the original change, so I didn’t fully catch the broader alignment you were aiming for with the Golang version. Appreciate your patience, and let me know if anything else needs tweaking!

tomaszbmf commented:
@AntoxaAntoxic @CTMBNara Just following up on the requested changes. Let me know if anything else is needed to move forward with merging.
